plugins/modules/mongodb_replicaset.py

```diff
@@ -433,16 +433,15 @@ def main():
     except Exception as e:
         module.fail_json(msg='Unable to connect to database: %s' % to_native(e))
 
-    mongo_auth(module, client)
-
     try:
-        rs = replicaset_find(client)
+        rs = replicaset_find(client)  # does not require auth
     except Exception as e:
         module.fail_json(msg='Unable to connect to query replicaset: %s' % to_native(e))
 
     if isinstance(rs, str):
         if replica_set == rs:
             if reconfigure:
+                mongo_auth(module, client)
```

Comment:
I think you also need `mongo_auth` just before the `replicaset_add` call below.

Reply:
Can you actually enable auth before initiating a replicaset? It's been such a long time I don't really remember.

Reply:
Yes... https://docs.mongodb.com/manual/tutorial/deploy-replica-set-with-keyfile-access-control/

Reply:
I think this is correct. Resolving.